### ISSUE TITLE
Accuracy topk-1 and top-5 metrics are incorrect

### ISSUE THREAD

**Issue by laclouis5** [riccardomusmeci/mlx-image#15]
The accuracy values computed as part of the benchmark are likely wrong for two reasons:

* The result varies when the batch size is changed, which should not be the case.
* Results from `torchmetrics` (which I trust more) is different (accuracy is around 1% lower).

I did not investigated too deep the actual error since the best solution may be to simply use a more reliable library for computing the metrics.

I recommend using either `torchmetrics` or `sklearn` for this purpose. The `torchmetrics` implementation is straightforward (`update` calls in the loop and a `compute` call at the end) and the `sklearn` implementation requires a bit more work but does not depend on PyTorch tensors (NumPy only).

**Comment by riccardomusmeci** [comment]
Working on it with sklearn. I want mlx-image as free as possible from torch. Thanks for reporting the issue. Planning on pushing the fix as soon as I re-run the experiments.
